### DISCUSSION TITLE
Put a link on a picture?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [py-pdf/fpdf2#769] [General]
Hello - i try to import an image and put a link on the image with the following code and the attached files:

![image](https://user-images.githubusercontent.com/57018730/232253785-bae56127-93e4-45d8-90ed-1188da293b70.jpg)

```
from fpdf import FPDF

pdf = FPDF()
pdf.add_page()
pdf.set_font("helvetica", size=36)
line_height = 10
text = "Text link"
pdf.text(x=0, y=10, txt="Text link")
width = pdf.get_string_width(text)
pdf.link(x=0, y=0, w=width, h=line_height, link="https://github.com/PyFPDF/fpdf2")
pdf.image("image.jpg")
pdf.output("hyperlink.pdf")
```

When i create the pdf - see attached
[hyperlink.pdf](https://github.com/PyFPDF/fpdf2/files/11240699/hyperlink.pdf)
the link is allways before the image.

Is it somehow possible to put the link directly on the image - if possibly centered in the middle?

**Comment by gmischler** [comment]
Hi @Rapid1898-code ,

As stated on https://pyfpdf.github.io/fpdf2/Links.html#hyperlink-with-fpdflink:

"The `FPDF.link()` is a low-level method that defines a **rectangular clickable area**."

if you look at this line in your code:
```python
    pdf.link(x=0, y=0, w=width, h=line_height, link="https://github.com/PyFPDF/fpdf2")
```

You may notice the **x, y, w, and h** arguments. They allow you to exactly specify the clickable area on the page.

**Comment by Rapid1898-code** [comment]
Hello - thanks to the reply - i changed the code to this:

`pdf.link(x=100, y=100, w=width, h=line_height, link="https://github.com/PyFPDF/fpdf2")`

But the text is still before the image.
I want to have the link with the text on the image.

**Reply by Lucas-C** [reply]
You shoud adjust the `x` & `y` coordinates passed to both `FPDF.text()` and `FPDF.link()` in order to position those elements on the page:
https://en.wikipedia.org/wiki/Cartesian_coordinate_system
```python
from fpdf import FPDF

pdf = FPDF()
pdf.add_page()
pdf.set_font("helvetica", size=36)
pdf.image("image.jpg")
text = "Text link"
text_x = 50
text_y = 50
pdf.text(x=text_x, y=text_y, txt=text)
width = pdf.get_string_width(text)
pdf.link(x=text_x, y=text_y, w=width, h=pdf.font_size, link="https://github.com/PyFPDF/fpdf2")
pdf.output("discuss_769.pdf")
```

**Comment by EminAbdullayev1998** [comment]
```python
from fpdf import FPDF

pdf = FPDF()
pdf.add_page()

# Add image
pdf.image("image.jpg", x=10, y=10, w=100)

# Add link over image
pdf.set_xy(x=10, y=10)
pdf.set_text_color(0, 0, 255)
pdf.set_font("Arial", size=12)
pdf.cell(w=100, h=30, txt="Click here to go to GitHub", ln=0, align="C", link="https://github.com/PyFPDF/fpdf2")

pdf.output("hyperlink.pdf")
```

**Reply by Lucas-C** [reply]
Yes, it's a very good suggestion to use `.cell()` instead of `.text()` + `.link()`

Thanks @EminAbdullayev1998!

**Comment by Rapid1898-code** [comment]
Hello - thanks a lot to both of you.
Finally achieved what i want to do.
